ARCHITECTURE.md

```diff
@@ -0,0 +1,127 @@
+
+Kedro-Viz can exist either as:
+
+- A standalone web app, which is [published to PyPI](https://pypi.org/project/kedro-viz/) and can be run as a Kedro plugin from the CLI
+- A React component, which is [published to npm](https://www.npmjs.com/package/@quantumblack/kedro-viz) and can be imported into a larger React application
+
+To allow the Kedro-Viz web app to be used as a Kedro plugin, first the JavaScript app is compiled into a static build, then it is bundled with a simple Python server and [published to PyPI](https://pypi.org/project/kedro-viz/).
+
+## Component package/library
+
+To publish Kedro-Viz as a React component library, it is first transpiled to the `/lib` directory with Babel. This process requires that the web worker be fully compiled (including its dependencies) with webpack, as it exists in a separate context requiring custom webpack loaders, which cannot be relied upon in an external parent application.
+
+When you import Kedro-Viz from npm, you can pass pipeline data to the component via the `data` prop:
+
+```jsx
+<KedroViz
+  data={{ nodes: [...], edges: [...], ... }}
+  theme="dark" />
+```
+
+## Data sources
+
+On initialisation, the app uses a string data token (e.g. 'json' or 'animals') to [determine the data source](CONTRIBUTING.md#data-sources).
+
+You can find example datasets in [/src/utils/data/](/src/utils/data/), which illustrate the basic API structure.
+
+## Synchronous data loading
+
+Some data source tokens instruct the app to synchronously `import` [test](/src/utils/data/animals.mock.json)/[demo](/src/utils/data/demo.mock.json) data from JSON files in the `/src/utils/data` directory, or to generate it randomly on page-load. Random data can be seeded with a 'seed' query string in the URL, to allow randomly-generated layouts to be replicated.
+
+## Asynchronous data loading
+
+Kedro-Viz loads data asynchronously in production from the API, or when using the 'json' data source identifier in development. The API provides two types of data source: pipeline endpoints and node endpoints.
+
+### Pipeline API endpoints
+
+Each pipeline endpoint corresponds to a different registered pipeline. Only one registered pipeline should be loaded at a time, so loading data from a pipeline endpoint will reset the pipeline state in the store. Each pipeline dataset contains all the data required to render the graph.
+
+On first page-load, the app always loads the `/api/main` endpoint first. This is the endpoint that corresponds to the 'default' pipeline. The app can load other pipelines from `/api/pipeline/<id>`. If another pipeline is saved as the active pipeline in `localStorage`, and if it exists in the current project, then the app will load that pipeline on first page load, but it will always load the `/api/main` endpoint first, in order to check whether the active pipeline is present at that endpoint before requesting it.
```

Comment:
Similarly could clarify "active pipeline" as "user's active pipeline"? Could the last sentence be split in two to make it a bit simpler?